### PR TITLE
Note Entity API changes

### PR DESCRIPTION
**Pull request by gbuisson**
This PR accommodates for the changes in the CTIM Note entity, namely:

- remove `entity_id` and introduce `related_entities` in the schema
- introduce `note_class` in the schema
- updated the entity search query parameters

<a name="qa">[§](#qa)</a> QA
============================

You can test the API following all the sample calls in the design documentation

  

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
public: updated Note Entity API
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

**Review by yogsototh — APPROVED** [review]
LGTM!

File: src/ctia/entity/note/schemas.clj [hunk 1]
```diff
@@ -47,11 +49,13 @@
 (def searchable-fields
   #{:id
     :source
-    :entity_id
+    :related_entities.entity_id
+    :related_entities.entity_type
+    :note_class
```
Comment (ereteog) [inline src/ctia/entity/note/schemas.clj:54]: searchable fields are for fields that we want to sort on by default, without actively filtering on it. I do not think that's desired here.

File: src/ctia/stores/es/mapping.clj [hunk 2]
```diff
@@ -256,6 +256,12 @@
     :columns {:enabled false}
     :rows {:enabled false}}})
 
+(def note-related-entity
+  {:type "object"
+   :properties
+   {:entity_type token
+    :entity_id token}})
```
Comment (ereteog) [inline src/ctia/stores/es/mapping.clj:263]: We tend to put entity specific mappings in the corresponding ns.

File: src/ctia/entity/note/schemas.clj (hunk 1 above)
Comment (gbuisson) [reply src/ctia/entity/note/schemas.clj]: I removed the unneeded fields

File: src/ctia/stores/es/mapping.clj (hunk 2 above)
Comment (gbuisson) [reply src/ctia/stores/es/mapping.clj]: done

**Pull request merged**